src/models/transaction.ts

```diff
@@ -292,7 +292,10 @@ class Transaction {
       txSize += 64;
     }
 
-    let sumOutputs = this.getOutputsSum();
+    // Here we have to explicitly convert a bigint to a number (a double), which loses precision,
+    // but is exactly compatible with the reference Python implementation because of the calculations below.
+    let sumOutputs = Number(this.getOutputsSum());
```

Comment:
We could keep the `sumOutputs` variable as a `BigInt`, but that wouldn't be useful because its only purpose is to calculate the `amount` below, which is then used to calculate the weight:

https://github.com/HathorNetwork/hathor-wallet-lib/blob/107d93e76c374d570e5e7227d49b8bb3a41f67e0/src/models/transaction.ts#L297-L309

Since the `weight` is a `Number` and the constants used to calculate it are too, `amount` must also be a `Number`, otherwise JS would error (`TypeError: Cannot mix BigInt and other types, use explicit conversions`).

We have the same problem in Python, but Python does the conversion automatically, so we don't have to do it explicitly:

https://github.com/HathorNetwork/hathor-core/blob/4da0fb9186793ee1116eab39915441909fcf1c28/hathor/daa.py#L201-L208

In that case, `tx.sum_outputs` is an `int` but `amount` is a `float`.

This does mean that we lose precision, but as stated in code comment, this is compatible with the weight calculated by the full node.